### PR TITLE
Datum antag refactor

### PR DESCRIPTION
**Pull request by 4dplanner**
This adds the ability to create a datum-based antag as a group, currently used for IAA and the document exchange objective. This allows for less dependence on gamemode code, which brings us closer to DATUM HEAVEN.

### CODE REVIEW

**Comment by Cyberboss** [comment]
>DATUM HEAVEN

**Comment by Xhuis** [comment]
We've come so far into datum antags that we're remaking the remake that we already have in place.

**Comment by optimumtact** [comment]
It's a good sign we're evolving the design to suit better

**Comment by Xhuis** [comment]
I get it, of course; I just find it kind of funny that we're already trying to apply a fresh coat of paint on top of the new fresh coat of paint that hadn't actually finished drying, or been painted on the rest of the house.

**Comment by Cyberboss** [comment]
>Initialize

**Comment by AnturK** [comment]
I mean this is okay but i believe for this to be useful you'd need these teams in some explicit datum.

File: code/datums/antagonists/antag_datum.dm [hunk 1]
```diff
@@ -61,3 +65,9 @@
 
 /datum/antagonist/proc/farewell()
 	return
+
+/datum/antagonist/proc/create_antagonist_group(var/list/datum/mind/targets) //overridden where creating a group is different, e.g double agents (internal affairs)
```
Comment (RemieRichards) [inline code/datums/antagonists/antag_datum.dm]: dont put ```var/``` in proc args.

**Comment by 4dplanner** [comment]
@AnturK: I'm sorry, I don't quite understand what you mean - I'll try knocking together an example to add to this PR later today to show how it can be used.

**Comment by 4dplanner** [comment]
Closing this until I have a proper showcase

**Pull request closed without merging**